core/src/com/group66/game/cannon/BallManager.java

```diff
@@ -340,22 +333,22 @@ public void draw(SpriteBatch batch, float delta) {
         
         /* Draw shot ball */
         for (Ball ball : ballList) {
-            ball.draw(batch, delta);
+            ball.draw(batch, ball.getType().getAnimation(), delta);
         }
 
         /* Draw static target balls */
         for (Ball ball : ballStaticList) {
-            ball.draw(batch, delta);
+            ball.draw(batch, ball.getType().getAnimation(), delta);
         }
 
         /* Draw popping balls */
         for (Ball ball : ballPopList) {
-            ball.draw(batch, delta);
+            ball.draw(batch, null, delta);
```

Comment:
The poging animation has a start and stop time, so this one is added and saved when the startPopping  is called. Here we want to continue with that animation, so we pass null as the argument won't be used.

I could change this so there is another draw in Ball which does not take an animation argument.